src/router/AppRouter.tsx

```diff
@@ -1,19 +1,35 @@
-import { Route, Routes } from "react-router-dom";
+import { Route, Routes, Navigate } from "react-router-dom";
 import { MainTemplate } from "../components/MainTemplate/MainTemplate";
-import { Account } from "../pages/account";
-import { GamePage } from "../pages/game";
-import { HomePage } from "../pages/home";
-import { SignUpPage } from "../pages/signup";
+import { Account } from "../pages/Account";
+import { BlackJack } from "../pages/BlackJack";
+import { Home } from "../pages/Home/Home";
+import { SignUp } from "../pages/SignUp";
 import { routes } from "../routes/routes";
+import { useAppSelector } from "../store/hooks";
+import { getUserInfo } from "../store/selectors";
 
 export const AppRouter = () => {
+  const { isAuthorized } = useAppSelector(getUserInfo);
   return (
     <Routes>
       <Route path={routes.HOME} element={<MainTemplate />}>
-        <Route index element={<HomePage />} />
-        <Route path={routes.SIGNUP} element={<SignUpPage />} />
-        <Route path={routes.GAME} element={<GamePage />} />
-        <Route path={routes.ACCOUNT} element={<Account />} />
+        <Route
+          index
+          element={!isAuthorized ? <Navigate to={routes.SIGNUP} /> : <Home />}
```

Comment:
need to think about how to improve it. What if we will have 50 protected routes? On every protected route you will need to write ternary operator and <Navigate .... />